### PR TITLE
Raise min phpcs `php_version` to 8.2

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/Roave/BetterReflection/pull/1423#issuecomment-2068114870

### CODE REVIEW

File: src/Reflection/Adapter/ReflectionIntersectionType.php [hunk 1]
```diff
@@ -35,7 +35,6 @@ public function __toString(): string
         return $this->betterReflectionType->__toString();
     }
 
-    /** @return false */
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionIntersectionType.php:38]: This stricter type is intentional 🤔

File: src/Reflection/ReflectionIntersectionType.php [hunk 2]
```diff
@@ -53,7 +53,6 @@ public function getTypes(): array
         return $this->types;
     }
 
-    /** @return false */
```
Comment (Ocramius) [inline src/Reflection/ReflectionIntersectionType.php:56]: This stricter type is intentional 🤔

File: src/Reflection/Adapter/ReflectionIntersectionType.php (hunk 1 above)
Comment (staabm) [reply src/Reflection/Adapter/ReflectionIntersectionType.php:38]: it is removed by phpcs,.. I guess we can't change the native type because BC?

Comment (Ocramius) [reply src/Reflection/Adapter/ReflectionIntersectionType.php:38]: We can 👍

Comment (staabm) [reply src/Reflection/Adapter/ReflectionIntersectionType.php:38]: done

**Review by Ocramius — COMMENTED** [review]
Groan:

```
There was 1 error:

1) Roave\BetterReflectionTest\Reflection\Adapter\ReflectionIntersectionTypeTest::testAdapterMethods with data set #1 ('allowsNull', null, true, [])
PHPUnit\Framework\MockObject\IncompatibleReturnValueException: Method allowsNull may not return value of type bool, its declared return type is "false"

/home/runner/work/BetterReflection/BetterReflection/test/unit/Reflection/Adapter/ReflectionIntersectionTypeTest.php:61

ERRORS!
Tests: 13152, Assertions: 73866, Errors: 1, Skipped: 1.
```

I think our test is too strict here.

Wdyt, @kukulich?

**Comment by staabm** [comment]
hmm I think the test is buggy.. I wonder why it worked before

**Comment by Ocramius** [comment]
I think it expects exactly the same types, although we intentionally do stricter types here

**Review by Ocramius — APPROVED** [review]
Thanks @staabm!

**Pull request merged**